### ISSUE TITLE
Model contributes nothing in the live path: train/serve feature-scaling skew collapses every score to ~0.731

### ISSUE THREAD

**Issue by Ayyankhan101** [Ayyankhan101/Microguard#19]
## Summary

On realistic live traffic the micrograd model does not discriminate at all: it returns **~0.731 for every request**, bot or human, identical to three decimals. All working detection comes from the heuristic labeler; the blended score inherits the heuristic's separation and the model adds nothing. This was measured with the new local realism lab (`scripts/lab/`, added in #18).

The model file itself is fine. Fed correctly scaled inputs directly it discriminates:

| input | `BotDetector.predict` |
|---|---|
| all zeros | 0.02 |
| human-shaped vector | 0.04 |
| bot-shaped vector | 0.66 |
| all ones | 0.73 |

The problem is train/serve **feature-scaling skew**, not the weights.

## Root cause

`BotDetector.normalize()` (`microguard/model.py:107`) min-max scales each feature using ranges in `data/normalization.json`. Those ranges were computed from **complete multi-request sessions** in the training scan, so the maxes are large:

```
maxs (excerpt): requests_per_minute_1m/5m = 204, time_since_last_request = 727.5,
                time_since_session_start = 1_224_575, request_count-like = 19_917
```

In the live path (`microguard/live/scorer.py`), the scorer scores **each request as the session accumulates**, so most requests are scored when the session still has one or a few requests and the raw feature values sit near the bottom of those training ranges. After `(value - min) / (max - min)` almost every column collapses to ~0. The model then receives a near-constant vector for every actor and outputs its bias-dominated constant, ~0.731.

Confirmed by feeding real archived live feature vectors back through the model: widely varying raw vectors (6046 distinct for bots) all normalize into a narrow near-zero band and all predict 0.7309–0.7310.

## Evidence (from the lab)

45 human sessions + 39 bots (real ffuf/sqlmap/nuclei + scrapers), scored against exact ground truth:

- Model, every threshold: flags all actors below ~0.73, none above — no threshold separates the classes. Per-class model score: bot mean 0.731, human mean 0.731.
- Blend at the shipped default (> 0.85): 39/39 bots caught, 0/45 humans flagged — but that separation is entirely the heuristic labeler's, floored/capped into the blend.

Reproduce: `bash scripts/lab/run.sh` and read the scorecard.

## Why it matters

The model is dead weight in production today. Either it should be made to work or the system should lean on the heuristics openly, rather than presenting a blended score that implies the model is contributing.

## Options

1. **Fix the scaling.** Compute normalization ranges from the same incremental, per-request feature distribution the live scorer produces (or scale features on a distribution that matches serve time), so live vectors land inside the trained range. This is the train/serve-skew fix.
2. **Retrain on real sessions.** The live evaluation (`docs/howto-evaluate-on-live-traffic.md`) is designed to produce real human sessions; retrain with normalization derived from that same feature pipeline.
3. **Robust scaling.** Replace min-max with a scaler less sensitive to the multi-request tail (e.g. per-feature standardization or quantile clipping), which would tolerate the range mismatch.

## Acceptance

- On the lab traffic, the model alone separates the classes better than chance (some threshold with meaningfully higher bot recall than human FP), measured by `scripts/lab/scorecard.py`.
- A regression test asserts the model does not return a single constant across a set of clearly-different live-shaped feature vectors.

Related: #18 (lab + model-score archiving), the existing train/serve normalization work (commit 885cd62), and the standing concern that the model has no real human training data.

**Comment by Ayyankhan101** [comment]
## Prototype confirms the fix (PR #20)

Retrained the tiny MLP on live-shaped feature vectors from a lab archive, with normalization computed from that same live distribution (`scripts/lab/retrain_from_lab.py`). Trained on one lab run, tested on a **fresh** run it never saw.

| detector | recall (bots) | human FP | F1 |
|---|---|---|---|
| shipped model (any threshold) | flags all or none — no separation | — | 0.00 |
| retrained model > 0.50 | 51/51 (100%) | 0/60 (0%) | 1.00 |

Score spread went from a collapsed constant (~0.731 for every actor) to a clean split: bots ~0.73, humans ~0.27. So option 1 in the issue (normalize on the live per-request distribution and retrain) is sufficient to make the model discriminate.

Two things the real fix still has to settle, out of scope for the prototype:

1. **Threshold interaction.** The retrained model's output tops out near 0.73, below the 0.85 block default, so the model's own score can't drive blocking without retuning the threshold or the training target.
2. **Generalization.** Both runs use the same lab traffic generator, so this proves the mechanism, not that it holds for real internet humans. Validate with the live EC2 run (`docs/howto-evaluate-on-live-traffic.md`) before shipping a retrained model.
